build.gradle

```diff
@@ -52,43 +52,50 @@ subprojects {
+    BUILD_PLAN: 'com.google.cloud.tools:jib-build-plan:0.3.1',
+    EXTENSION_COMMON: 'com.google.cloud.tools:jib-plugins-extension-common:0.2.0',
+    GRADLE_EXTENSION: 'com.google.cloud.tools:jib-gradle-plugin-extension-api:0.3.0',
+    MAVEN_EXTENSION: 'com.google.cloud.tools:jib-maven-plugin-extension-api:0.3.0',
+
+    COMMONS_COMPRESS: 'org.apache.commons:commons-compress:1.19',
+    JACKSON_DATABIND: 'com.fasterxml.jackson.core:jackson-databind:2.11.0',
+    JACKSON_DATAFORMAT_YAML: 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.11.0',
+    ASM: 'org.ow2.asm:asm:7.3.1',
+    PICOCLI: 'info.picocli:picocli:4.2.0',
+
+    MAVEN_API: 'org.apache.maven:maven-plugin-api:3.5.2',
+    MAVEN_CORE: 'org.apache.maven:maven-core:3.5.2',
+    MAVEN_COMPAT: 'org.apache.maven:maven-compat:3.5.2',
+    MAVEN_ANNOTATIONS_LATEST: 'org.apache.maven.plugin-tools:maven-plugin-annotations:3.6.0',
+    MAVEN_ANNOTATIONS_OLD: 'org.apache.maven.plugin-tools:maven-plugin-annotations:3.5',
+    MAVEN_TESTING_HARNESS: 'org.apache.maven.plugin-testing:maven-plugin-testing-harness:3.3.0',
+    MAVEN_VERIFIER: 'org.apache.maven.shared:maven-verifier:1.7.2',
 
-    COMMONS_COMPRESS: "org.apache.commons:commons-compress:1.19",
-    JACKSON_DATABIND: "com.fasterxml.jackson.core:jackson-databind:2.11.0",
-    JACKSON_DATAFORMAT_YAML: "com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.11.0",
-    ASM: "org.ow2.asm:asm:7.3.1",
-    PICOCLI: "info.picocli:picocli:4.2.0",
 
-    MAVEN_API: "org.apache.maven:maven-plugin-api:3.5.2",
-    MAVEN_CORE: "org.apache.maven:maven-core:3.5.2",
-    MAVEN_COMPAT: "org.apache.maven:maven-compat:3.5.2",
 
     //test
-    JUNIT: "junit:junit:4.13",
-    MOCKITO_CORE: "org.mockito:mockito-core:3.2.4",
-    SLF4J_API: "org.slf4j:slf4j-api:1.7.25",
-    SYSTEM_RULES:  "com.github.stefanbirkner:system-rules:1.19.0",
-    JBCRYPT: "org.mindrot:jbcrypt:0.4",
+    JUNIT: 'junit:junit:4.13',
+    MOCKITO_CORE: 'org.mockito:mockito-core:3.2.4',
+    SISU_PLEXUS: 'org.eclipse.sisu:org.eclipse.sisu.plexus:0.3.4',
+    SLF4J_API_OLD: 'org.slf4j:slf4j-api:1.7.25',
+    SLF4J_API_LATEST: 'org.slf4j:slf4j-api:1.7.30',
+    SLF4J_SIMPLE: 'org.slf4j:slf4j-simple:1.7.30',
+    SYSTEM_RULES:  'com.github.stefanbirkner:system-rules:1.19.0',
+    JBCRYPT: 'org.mindrot:jbcrypt:0.4',
   ]
 
   // Use this to ensure we correctly override transitive dependencies
   // TODO: There might be a plugin that does this
   task ensureTransitiveDependencyOverrides {
-    def rules = [
-        'google-http-client': dependencyStrings.GOOGLE_HTTP_CLIENT.split(':')[-1],
-        'google-http-client-apache-v2': dependencyStrings.GOOGLE_HTTP_CLIENT_APACHE_V2.split(':')[-1],
-    ]
+    def dependenciesList = [dependencyStrings.GOOGLE_HTTP_CLIENT, dependencyStrings.GOOGLE_HTTP_CLIENT_APACHE_V2]
+    def rules = dependenciesList.collectEntries{[/*name: */ it.split(':')[1], /*version: */ it.split(':')[-1]]}
```

Comment:
I think using `[2]` instead of `[-1]` will help us catch bugs better?

Reply:
Good point, done.